backend/templates/create-report/template-files/routes/index.js

```diff
@@ -1,21 +1,18 @@
-const controller = require("@controllers/reports/$reportName.js");
+const getControllerFrom = require("@controllers/reports/genericReportController");
 // const validators = require("@validators/report");
-const what = "report";
 
-const routes = [
-  {
-    method: "GET",
-    url: `/${what}/projects/$reportName`,
-    onRequest: controller.$reportName,
-    handler: controller.getReport,
-  },
-];
-const registerRoutes = (fastify, options, done) => {
-  // Ensure all of the routes above get registered.
-  routes.forEach((route) => fastify.route(route));
-  done();
-};
+const name = `$reportName`;
+const controller = getControllerFrom(name);
 
 module.exports = {
-  registerRoutes,
+  registerRoutes: (fastify, options, done) => {
+    fastify.route({
+      method: "GET",
+      url: `/report/projects/${name}`,
+      // schema: getAll,
+      onRequest: controller[name],
+      handler: controller.getReport,
+    });
+    done();
+  },
```

Comment:
Would it be possible to get this down to one route for all reports in the same way we now have one controller? It looks like we're nearly there already here. So instead of getting the controller name from the route, we get it from a query param and that query param is used in the generic controller to choose the correct model. That's beyond the scope of this ticket but it would be cool.

Reply:
I think it's possible. I'll explore that in a future ticket - this one is already pretty complicated, but yeah that's a cool idea.